### ISSUE TITLE
MIGBuilder not working with FeatureModelCnf

### ISSUE THREAD

**Issue by mnieke** [FeatureIDE/FeatureIDE#1351]
#### Prerequisitives
* **Eclipse Version:**  2021-06
* **FeatureIDE-Version:**  3.7.2
* **Operating system:** Windows 10
* **Java runtime used to run eclipse:** JVM 16
* **FeatureIDE Composer:**
* **Git branch/Commit-ID:**
* **Java compiler used to build the FeatureIDE plugin:**

### Issue description
We tried to create a model implication graph (MIG) using a FeatureModelCnf of a FeatureModel as parameter "satInstance". The resulting MIG was empty.
We know that the FeatureModelFormula class is the right one to use, but a student first used the FeatureModelCnf (which seems natural for a Cnf of a FeatureModel).
Attached the minimal code.
[Main.txt](https://github.com/FeatureIDE/FeatureIDE/files/8714756/Main.txt)

**Comment by tthuem** [comment]
@mnieke Thanks for your report. Just to double check, you have used the version that is almost one year old? Have you tried the current version already?

**Comment by mnieke** [comment]
Well indeed, I just checked it using the old version. If I find the time today, I will try it with the new one as well.

**Comment by skrieter** [comment]
Hi @mnieke 
I just checked the code you provided.

It might be confusing, but it is not intended that you use the constructor of FeatureModelCNF directly. In the third to last line of you code example, you are supposed to call the following:
`final CNF fmcnf = new FeatureModelFormula(featureModel).getCNF();`
The idea behind this is that the FeatureModelFormula object caches intermediate results. Thus, if you use it multiple times, then the CNF is not computed more than once.

Nevertheless, I just made a commit that hopefully clarifies things a bit through the code API. In the next version, the constructor of FeatureModelCNF will no longer be visible, but you will be able to execute the following, which will give you your desired result:
`final CNF fmcnf = FeatureModelCNF.fromFeatureModel(featureModel);`
